lib/screens/calendar/calendar.dart

```diff
@@ -0,0 +1,841 @@
+        }
+      }
+      return result;
+    } else
+      return List.filled(0, DateTime.now());
+  }
+
+  bool _isSameWeekAsCurrentWeek(DateTime date) {
+    return _getWeekNumber(date).toString() ==
+        _getWeekNumber(DateTime.now()).toString();
+  }
+
+  /// Calculates number of weeks for a given year as per https://en.wikipedia.org/wiki/ISO_week_date#Weeks_per_year
+  int numOfWeeks(int year) {
+    DateTime dec28 = DateTime(year, 12, 28);
+    int dayOfDec28 = int.parse(DateFormat("D").format(dec28));
+    return ((dayOfDec28 - dec28.weekday + 10) / 7).floor();
+  }
+
+  /// Calculates week number from a date as per https://en.wikipedia.org/wiki/ISO_week_date#Calculation
+  int _getWeekNumber(DateTime date) {
+    int dayOfYear = int.parse(DateFormat("D").format(date));
+    int woy = ((dayOfYear - date.weekday + 10) / 7).floor();
+    if (woy < 1) {
+      woy = numOfWeeks(date.year - 1);
+    } else if (woy > numOfWeeks(date.year)) {
+      woy = 1;
+    }
+    return woy;
+  }
+
+  String convertToIso8601(DateTime time) {
+    //need to have a manual convert because the default method have a special 's' -> :sssZ
+    return DateFormat("yyyy-MM-ddTHH:mm:ss'Z'").format(time);
+  }
+
+  String convertToIso8601inDays(DateTime time) {
+    String timeIn8601 = DateFormat("yyyy-MM-ddTHH:mm:ss'Z'").format(time);
+    return timeIn8601.substring(0, 10);
+  }
+
+  List<String> expandStartNEndDate(String first, String last) {
+    /// Returns a list of [DateTime] objects from [first] to [last], inclusive.
+    DateTime startDate = DateTime.parse(first);
+    DateTime endDate = DateTime.parse(last);
+    final dayCount = endDate.difference(startDate).inDays + 1;
+    // var length = (endDate.day != startDate.day)
+    //     ? endDate.difference(startDate).inDays + 1
+    //     : 1;
+
+    return List.generate(
+        dayCount,
+        (index) =>
+            convertToIso8601inDays(startDate.add(Duration(days: (index)))));
+
+    var result = List.generate(dayCount, (index) {
+      return convertToIso8601inDays(startDate.add(Duration(days: (index))));
+    });
+    return result;
+  }
+
+  loadingInputPeriodData() async {
+    var from = convertToIso8601(_focusedDay!.subtract(Duration(days: 31)));
+    var to = convertToIso8601(_focusedDay!.add(Duration(days: 31)));
+    final client = ref.read(gqlClientProvider);
+
+    _calendarPeriod = client
+        .query$CalendarPeriod(Options$Query$CalendarPeriod(
+      variables: Variables$Query$CalendarPeriod(from: from, to: to),
+    ))
+        .onError((error, stackTrace) {
+      1;
+      throw Error();
+    }).then((value) {
+      if (value.hasException) {
+        throw ErrorDescription(value.exception.toString());
+      }
+      List<String> activeDaysList = [];
+      value.parsedData?.calendar?.period.activeDays
+          .forEach((element) => activeDaysList.add(element));
+      activeDaysList.forEach((dayTime) {
+        activeDaysPeriod.add(DateTime.parse(dayTime));
+      });
```

Comment:
This can be simplified with map(). See https://stackoverflow.com/questions/49941361/dart-mapping-a-list-list-map

```dart
var activeDays = value.parsedData?.calendar?.period.activeDays.map((element) => DateTime.parse(element));
setState(() {
    activeDaysPeriod = HashSet.from(activeDays);
})
```

Reply:
fixed